core/src/test/java/org/elasticsearch/discovery/zen/ElectMasterServiceTests.java

```diff
@@ -69,37 +89,54 @@ public void testSortByMasterLikelihood() {
             }
             prevNode = node;
         }
+    }
+
+    public void testTieBreakActiveMasters() {
+        List<DiscoveryNode> nodes = generateRandomCandidates().stream().map(Candidate::getNode).collect(Collectors.toList());
+        DiscoveryNode bestMaster = electMasterService().tieBreakActiveMasters(nodes);
+        for (DiscoveryNode node: nodes) {
+            if (node.equals(bestMaster) == false) {
+                assertTrue(bestMaster.getId().compareTo(node.getId()) < 0);
+            }
+        }
+    }
 
+    public void testHasEnoughNodes() {
+        List<DiscoveryNode> nodes = rarely() ? Collections.emptyList() : generateRandomNodes();
+        ElectMasterService service = electMasterService();
+        int masterNodes = (int) nodes.stream().filter(DiscoveryNode::isMasterNode).count();
+        service.minimumMasterNodes(randomIntBetween(-1, masterNodes));
+        assertThat(service.hasEnoughMasterNodes(nodes), equalTo(masterNodes > 0));
+        service.minimumMasterNodes(masterNodes + 1 + randomIntBetween(0, nodes.size()));
+        assertFalse(service.hasEnoughMasterNodes(nodes));
+    }
+
+    public void testHasEnoughCandidates() {
+        List<Candidate> candidates = rarely() ? Collections.emptyList() : generateRandomCandidates();
+        ElectMasterService service = electMasterService();
+        service.minimumMasterNodes(randomIntBetween(-1, candidates.size()));
+        assertThat(service.hasEnoughCandidates(candidates), equalTo(candidates.size() > 0));
+        service.minimumMasterNodes(candidates.size() + 1 + randomIntBetween(0, candidates.size()));
+        assertFalse(service.hasEnoughCandidates(candidates));
     }
 
     public void testElectMaster() {
-        List<DiscoveryNode> nodes = generateRandomNodes();
+        List<Candidate> candidates = generateRandomCandidates();
         ElectMasterService service = electMasterService();
-        int min_master_nodes = randomIntBetween(0, nodes.size());
+        int min_master_nodes = randomIntBetween(0, candidates.size());
```

Comment:
While we are here, can we give this a proper Java variable name (`minMasterNodes`)?